mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -703,8 +703,28 @@ where
                         let mut internal_id_hash = index.to_le_bytes().to_vec();
                         internal_id_hash.extend_from_slice(tx_hash.as_bytes());
                         drop_mutability!(internal_id_hash);
+                        let len = internal_id_hash.len();
+                        // Discuss: This truncates `internal_id_hash` to 32 bytes instead of using all 33 bytes (index + tx_hash).
+                        // This is a limitation kept for backward compatibility. Changing to 33 bytes would
+                        // alter the internal_id calculation, causing existing wallets to see duplicate transactions
+                        // in their history. A proper migration would be needed to safely transition to using the full 33 bytes.
+                        let internal_id_hash: [u8; 32] = match internal_id_hash
+                            .get(..32)
+                            .and_then(|slice| slice.try_into().ok())
+                        {
+                            Some(hash) => hash,
+                            None => {
+                                log::debug!(
+                                    "Invalid internal_id_hash length for tx '{}' at index {}: expected 32 bytes, got {} bytes.",
+                                    tx_hash,
+                                    index,
+                                    len
+                                );
+                                continue;
+                            },
+                        };
```

Comment:
Discussion point

Reply:
Maybe, we could use the correct calculation if it's a fresh start and do the otherwise if there are DB files already. After a year or so, we could remove the old/wrong approach from the code base. How does that sound?